Dockerfile

```diff
@@ -3,26 +3,20 @@ MAINTAINER Tim Perry <pimterry@gmail.com>
 
 USER root
 
-RUN apt-get update && \
-    export DEBIAN_FRONTEND=noninteractive && \
-    # Set password to temp-password - reset to random password on startup
-    echo mysql-server mysql-server/root_password password temp-password | debconf-set-selections && \
-    echo mysql-server mysql-server/root_password_again password temp-password | debconf-set-selections && \   
-    # Instal MySQL for data, node as the JS engine for uglifier
-    apt-get install -y mysql-server nodejs
-    
-COPY . /opt/staytus
+ENV DEBIAN_FRONTEND="noninteractive"
 
-RUN cd /opt/staytus && \
-    bundle install --deployment --without development:test
-
-ENTRYPOINT /opt/staytus/docker-start.sh
+ADD docker-start.sh /docker-start.sh
+ADD . /opt/staytus/
 
-# Persists all DB state
-VOLUME /var/lib/mysql
+RUN chmod 755 docker-start.sh && \
+    apt-get update && \
+    # Instal node as the JS engine for uglifier
+    apt-get install -y nodejs && \
+    apt-get clean && \
+    rm -rf /var/lib/apt/lists/* /tmp/* /var/tmp/* && \
+    cd /opt/staytus && \
+    bundle install --deployment --without development:test
```

Comment:
Ordering is a interesting here.

In the original Dockerfile, all the generic setup (installing node, and the now-defunct mysql setup) was done in a separate preceding RUN step, before then COPYing and RUNing the parts that depend on content in the local directory.

Local development of the Dockerfile is much easier that way; you don't invalidate the old steps as you change content, so rebuilding the image is quite a bit quicker, and doesn't apt-get update from scratch and reinstall node again every time. This will work, but reorganising it just slightly makes local docker dev much easier.